pages/blogPosts/[slug].tsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import matter from 'gray-matter';
+
+import BlogPost from '../../components/BlogPost';
+
+import { Footer, Header } from "../../landing";
+
+import fs from 'fs';
+import { join } from 'path';
+
+const postsDirectory = 'content/blog/';
+
+
+const BlogPostPage = ({content, data}) => {
+  
+    return (
+        <>  
+            <Header/>
+            <BlogPost content={content} data={data}/>
+            <Footer bgColor="dark" fgColor="white"/>
+        </>
+    );
+};
+
+export async function getServerSideProps(context: { query: { slug: string; }; }) {
```

Comment:
Since blog post rarely change, and every time they change we need to do a new commit, we should use [`getStaticPops`](https://nextjs.org/docs/basic-features/data-fetching#getstaticprops-static-generation) and [`getStaticPaths`](https://nextjs.org/docs/basic-features/data-fetching#getstaticpaths-static-generation). Those functions run at build time and will make a better UX since it will be much more fast for the user.

Check this example for [reference](https://github.com/vercel/next.js/blob/canary/examples/blog-starter/pages/posts/%5Bslug%5D.js#L49).